virttest/test_setup.py

```diff
@@ -2079,3 +2081,46 @@ def disable_smt(params=None):
                 logging.debug("SMT turned off successfully")
             except process.CmdError, info:
                 raise exceptions.TestSetupFail("VM can not be started :%s" % info)
+
+
+class LibvirtdDebugLog(object):
+    """
+    Enable libvirtd log for testcase incase
+    with the use of param "enable_libvirtd_debug_log",
+    with additional params log level("libvirtd_debug_level")
+    and log file path("libvirtd_debug_file") can be controlled.
+    """
+    def __init__(self, env, test, log_level=1, log_file=""):
+        """
+        initialize variables
+
+        :param env: ENV object
+        :param test: Test object
+        :param level: debug level for libvirtd log
+        :param debug_file: debug file path
+        """
+        self.env = env
+        self.log_level = log_level
+        self.log_file = log_file
+        self.test = test
+        self.libvirtd = utils_libvirtd.Libvirtd()
+        self.libvirtd_conf = utils_config.LibvirtdConfig()
+        self.backupfile = "%s.bkup.debuglog" % self.libvirtd_conf.conf_path
+
+    def enable(self):
+        """ Enable libvirtd debug log """
+        if not self.log_file or not os.path.isfile(self.log_file):
```

Comment:
if user have given the log_file but file may or may not exist, if file doesn't exist then we can open a new file with user given log_file path?